### PR TITLE
[AppService] Fix #2258: Fixing issue where trying to create a webapp with certain runtimes was failing

### LINKED ISSUE #2258: vmss list_instance_connection_info should be list-instance-connection-info
This command should not be named with underscores and should probably be changed before GA of the module.

/cc @twitchax @johanste

### LINKED ISSUE #12258: Error creating an webapp with runtime option on v2.1.0
## Describe the bug

**Command Name**
`az webapp create`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:


update_site_config() takes 2 positional arguments but 3 were given
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 608, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 666, in _run_jobs_serially           results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 657, in _run_job
    cmd_copy.exception_handler(ex)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/commands.py", line 55, in _polish_bad_errors
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 636, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 306, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 493, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 135, in create_webapp     match['setter'](cmd, match, site_config)
TypeError: update_site_config() takes 2 positional arguments but 3 were given
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az webapp create -g $RESOURCE_GROUP -n $SITE_NAME -p $PLAN_NAME -r "java|1.8|Tomcat|8.5"`

note: not only for `java|1.8|Tomcat|8.5`

## Environment Summary
```
Linux-4.4.0-18362-Microsoft-x86_64-with-debian-stretch-sid
Python 3.6.5

azure-cli 2.1.0

```
## Additional Context

This is similar with #12229 .
But this happens on Azure CLI v2.1.0 and seems that this depends on runtime specified in the command.
I tried other runtimes, then creating with most of runtimes (except `node|x.y`) was failed.

### PR DESCRIPTION
**Pull request by panchagnula**
Fixes #12258 , #12247, #12229

**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by panchagnula** [comment]
@Juliehzl unfortunately your fix to fix the runtimes did not fix the issue with all commands so making a fix that should work with az webapp create as well. Can you take a look?

**Comment by yungezz** [comment]
this is a regression. @qwordy  @Juliehzl could you pls review the fix? thanks.

**Comment by qwordy** [comment]
You format the code. There is so many changes. Not easy to find real changes. Can you summarize the changes you made and mark the related lines? Thanks.

**Comment by Juliehzl** [comment]
> @Juliehzl unfortunately your fix to fix the runtimes did not fix the issue with all commands so making a fix that should work with az webapp create as well. Can you take a look?

LGTM. The change is the same as above. But can we add a test for `az webapp create` with --runtime parameter?

**Comment by panchagnula** [comment]
@Juliehzl we do have a test with NODE runtime which currently passes, I will add one with the other runtime/stack which currently fails. also is there a way to release this as a hot fix?

**Comment by panchagnula** [comment]
> You format the code. There is so many changes. Not easy to find real changes. Can you summarize the changes you made and mark the related lines? Thanks.

Ah sorry! didn't even realize that VSCode reformatted it on save - i need to get rid of that extension.
The changes are line 143 in custom.py and line 2350.  Thanks!

**Comment by panchagnula** [comment]
@qwordy thanks for pointing out the formatting changes - i update the file to include just the bug fix without the re-formatting. Hope this helps.
